src/main/scala/io/iohk/ethereum/nodebuilder/tooling/StorageConsistencyChecker.scala

```diff
@@ -0,0 +1,29 @@
+package io.iohk.ethereum.nodebuilder.tooling
+
+import com.typesafe.scalalogging.Logger
+
+import io.iohk.ethereum.db.storage.BlockHeadersStorage
+import io.iohk.ethereum.db.storage.BlockNumberMappingStorage
+
+object StorageConsistencyChecker {
+  type ShutdownOp = () => Unit
+
+  val DefaultStep = 1000
+
+  def checkStorageConsistency(
+      bestBlockNumber: BigInt,
+      blockNumberMappingStorage: BlockNumberMappingStorage,
+      blockHeadersStorage: BlockHeadersStorage,
+      shutdown: ShutdownOp,
+      step: Int = DefaultStep
+  )(implicit log: Logger): Unit =
+    Range(0, bestBlockNumber.intValue, step).foreach { idx =>
+      (for {
+        hash <- blockNumberMappingStorage.get(idx)
+        _ <- blockHeadersStorage.get(hash)
+      } yield ()).fold {
+        log.error("Database seems to be in inconsistent state, shutting down")
```

Comment:
Shouldn't we include information on how the user should act when this happens? I guess how to delete the DB?

Reply:
Is this a good place to add such documentation for the client? Not convinced, I think we have a mantis docs project.
We don't have a policy for such situations.

Reply:
In that case advising the user to check the website. But I think that saying nothing is the worse option :)

Reply:
I think we should log the missing block's hash and number. I'm not sure it would really helps us when this happens, but it does not hurt.